sdk/communication/Azure.Communication.Calling.Server/src/ConversationClient.cs

```diff
@@ -0,0 +1,622 @@
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="recordingId">The recording id to pause.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public virtual Response PauseRecording(string conversationId, string recordingId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ConversationClient)}.{nameof(PauseRecording)}");
+            scope.Start();
+            try
+            {
+                return RestClient.PauseRecording(conversationId, recordingId, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Resume recording of the call.
+        /// </summary>
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="recordingId">The recording id to pause.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public virtual async Task<Response> ResumeRecordingAsync(string conversationId, string recordingId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ConversationClient)}.{nameof(ResumeRecordingAsync)}");
+            scope.Start();
+            try
+            {
+                return await RestClient.ResumeRecordingAsync(conversationId, recordingId, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// resume recording of the call.
+        /// </summary>
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="recordingId">The recording id to resume.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public virtual Response ResumeRecording(string conversationId, string recordingId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ConversationClient)}.{nameof(ResumeRecording)}");
+            scope.Start();
+            try
+            {
+                return RestClient.ResumeRecording(conversationId, recordingId, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        private static T AssertNotNull<T>(T argument, string argumentName)
```

Comment:
We have added a PR that moves these into Azure.Core and we dont need to define them manually here. https://github.com/Azure/azure-sdk-for-net/pull/21542

Please remove and refer to Arguments.cs instead.

Reply:
Synced with master and remove those two methods.